test/utils/AmmPoolUtils.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.16;
+
+import "forge-std/Test.sol";
+import "../utils/TestConstants.sol";
+import "../../contracts/amm/AmmTreasury.sol";
+import "../../contracts/amm/AmmStorage.sol";
+import "../../contracts/oracles/IporOracle.sol";
+
+contract AmmPoolUtils is Test {
+    struct ExchangeRateAndPayoff {
+        uint256 initialExchangeRate;
+        uint256 exchangeRateAfter28Days;
+        uint256 exchangeRateAfter56DaysBeforeClose;
+        int256 payoff1After28Days;
+        int256 payoff2After28Days;
+        int256 payoff1After56Days;
+        int256 payoff2After56Days;
+    }
+
+    function calculateSoap(
```

Comment:
Don't need this because you can reuse _iporProtocol.ammSwapsLens.getSOAP(address(_iporProtocol.asset));

Reply:
you are right, changed